code/main.py

```diff
@@ -0,0 +1,351 @@
+
+@op(ins={"start": In(Nothing), "source": In(dagster_type=dict)})
+def summarize(context: OpExecutionContext, source: dict):
+    s3Minio = s3.MinioDatastore(
+        _pythonMinioAddress(GLEANER_MINIO_ADDRESS, GLEANER_MINIO_PORT), MINIO_OPTIONS
+    )
+    bucket = GLEANER_MINIO_BUCKET
+    source_name = "SOURCEVAL"
+    endpoint = _graphEndpoint()  # getting data, not uploading data
+    summary_namespace = _graphSummaryEndpoint()
+
+    try:
+        summarydf = get_summary4repoSubset(endpoint, source_name)
+        nt, g = summaryDF2ttl(summarydf, source_name)  # let's try the new generator
+        summaryttl = g.serialize(format="longturtle")
+        # Lets always write out file to s3, and insert as a separate process
+        # we might be able to make this an asset..., but would need to be acessible by http
+        # if not stored in s3
+        objectname = f"{SUMMARY_PATH}/{source_name}_release.ttl"  # needs to match that is expected by post
+
+        s3Minio.putTextFileToStore(summaryttl, S3ObjectInfo(bucket, objectname))
+        # inserted = sumnsgraph.insert(bytes(summaryttl, 'utf-8'), content_type="application/x-turtle")
+        # if not inserted:
+        #    raise Exception("Loading to graph failed.")
+    except Exception as e:
+        # use dagster logger
+        get_dagster_logger().error(f"Summary. Issue creating graph  {str(e)} ")
+        raise Exception(f"Loading Summary graph failed. {str(e)}")
+
+
+@op(ins={"start": In(Nothing), "source": In(dagster_type=dict)})
+def upload_summarize(context: OpExecutionContext, source: dict):
+    returned_value = post_to_graph(
+        "SOURCEVAL",
+        path=SUMMARY_PATH,
+        extension="ttl",
+        graphendpoint=_graphSummaryEndpoint(),
+    )
+    # the above can be done (with a better path approach) in Nabu
+    # returned_value = gleanerio(context, ("object"), "SOURCEVAL")
+    r = str("returned value:{}".format(returned_value))
+    get_dagster_logger().info(f"upload summary returned  {r} ")
+
+
+
+def harvest_with_source(source: GleanerSource):
+    @graph
+    def harvest():
+        """Harvest all assets for a given source. Since dagster cannot easily parametrize, we need to pass in source for each fn"""
+        get_dagster_logger().info(f"Harvesting source: {source}")
+
+        setup = gleaner(start = pull_docker_images(), source=source)
+
+        # # data branch
+        release = naburelease(start=setup, source=source)
+        upload = uploadrelease(start=release, source=source)
+        nabu_prune(start=upload, source=source)
+
+        # # prov branch
+        provdrain = nabu_provdrain(start=setup, source=source)
+        prune = nabu_provobject(start=provdrain, source=source)
+        provclear = nabu_provclear(start=prune, source=source)
+        nabu_provrelease(start=provclear, source=source)
+
+        # # org branch
+        orgs = nabuorgs(start=setup, source=source)
+        nabu_orgsrelease(start=orgs, source=source)
+
+    return harvest
```

Comment:
Trying to create a factory method that contains the graph fn with the proper source (CUAHSI, etc.) so it limits how much we need to dynamically create

Reply:
This pattern seems fine according to the docs https://dagster.io/blog/python-factory-patterns